lib/braveAlerter.js

```diff
@@ -0,0 +1,182 @@
+const express = require('express')
+const jsonBodyParser = require('body-parser').json()
+
+const AlertSession = require('./alertSession.js')
+const ALERT_STATE = require('./alertStateEnum.js')
+const AlertStateMachine = require('./alertStateMachine.js')
+const helpers = require('./helpers.js')
+const twilio = require('./twilio.js')
+
+class BraveAlerter {
+    constructor(getAlertSession, getAlertSessionByPhoneNumber, alertSessionChangedCallback, asksIncidentDetails, getReturnMessage) {
+        this.getAlertSession = getAlertSession
+        this.getAlertSessionByPhoneNumber = getAlertSessionByPhoneNumber
+        this.alertSessionChangedCallback = alertSessionChangedCallback
+
+        this.router = express.Router()
+        this.router.post('/sms', jsonBodyParser, this.handleTwilioRequest.bind(this))
```

Comment:
So right now ODetect uses `/sms` and Buttons uses `/messages`. I could take this in as a parameter to the constructor to have less disruption and fewer config changes, but I think that this shared library is going to end up with more endpoints as we build the app. So maybe it's best if they all stay consistent.

Reply:
I'd suggest picking a nice structure for the API and then patching any of the existing routes with a redirect until we update all of our 3rd party service integrations and all of our own devices.

Reply:
IMO there's no need to expose this through the library interface as we want the library to handle all this stuff going forward.

Reply:
Could you please explain a little more what you mean? By exposing it like this, the calling application can plunk these routes wherever it likes in its hierarchy (as described in the `README`).

Are we able to have more than one Express router within the same application? Because ODetect still has need of API endpoints that the sensors use.

Reply:
yes, you can mount multiple routers onto individual base routes and have them handle all the endpoints that are children of that base route.

we don't want the calling application to have to define every route that the BraveAlerter needs to implement. that's part of the work that we're trying to encapsulate in this library. therefore i don't think it's a good idea to take route definitions as parameters in the constructor. I could see an argument for taking a single base route as a parameter, but we may as well just collect the entire BraveAlerter API under `/alert` and reserve that for use by the library.